src/v/serde/protobuf/parser.h

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright 2024 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#include "bytes/iobuf.h"
+#include "container/chunked_hash_map.h"
+#include "container/fragmented_vector.h"
+
+#include <cstdint>
+
+namespace google::protobuf {
+class Descriptor;
+} // namespace google::protobuf
+
+namespace serde::pb {
+
+namespace parsed {
+
+struct message;
+
+/**
+ * A dynamic representation of a repeated field in a protobuf message.
+ */
+struct repeated {
+    std::variant<
+      chunked_vector<double>,
+      chunked_vector<float>,
+      chunked_vector<int32_t>, // Can be an enum value
+      chunked_vector<int64_t>,
+      chunked_vector<uint32_t>,
+      chunked_vector<uint64_t>,
+      chunked_vector<bool>,
+      chunked_vector<iobuf>, // string or bytes
+      chunked_vector<std::unique_ptr<message>>>
+      elements;
+};
```

Comment:
Yea this makes sense. I have an Iceberg value variant that represents lists as a vector of variants. I think it's a bit easier to implement, but there's no guarantee that each element has the same type as you get here

Reply:
Yeah I think technically this is faster too, which is why I dis this here. For maps there are too many combinations so I did the boxed version there (maps might be less common as well)